.github/workflows/deploy.yml

```diff
@@ -0,0 +1,154 @@
+
+    env:
+      # Define the instance information.
+      account-id: MIDEN_DEV_ACCOUNT_ID
+      oicdrole: midendev 
+      instance-id: ${{ inputs.network == 'testnet' && 'TESTNET_INSTANCE_TF' || 'DEVNET_INSTANCE_TF' }}
+
+      # Define the expected package names.
+      node-package:   miden-node-${{ inputs.gitref }}-${{ inputs.architecture }}.deb
+      faucet-package: miden-faucet-${{ inputs.gitref }}-${{ inputs.architecture }}.deb
+
+      # S3 path where packages are stored; used to send packages to instance as this isn't trivially possible directly.
+      s3-path: s3://release-artifacts-${{ secrets[env.account-id] }}
+        
+    steps:
+      # Checkout repo so we have access to the required workflow actions.
+      - name: Checkout repo
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      
+      # Download from github if its a version tag referece.
+      - name: Download packages from releases
+        if: ${{ startsWith(inputs.gitref, 'v') }}
+        run: |
+          gh release ${{ inputs.gitref }} download ${{ env.node-package }}
+          gh release ${{ inputs.gitref }} download ${{ env.node-package }}.checksum
+          gh release ${{ inputs.gitref }} download ${{ env.faucet-package }}
+          gh release ${{ inputs.gitref }} download ${{ env.faucet-package }}.checksum
+
+          sha256 --check ${{ env.node-package }}.checksum
+          sha256 --check ${{ env.faucet-package }}.checksum
+
+      # Otherwise build the packages from source.
+      #
+      # Note that we cannot build from the currently checked out repo source since that source
+      # defines our workflow actions, and not the compilation source target. For this reason we
+      # prefer building the binary using `cargo install ...`.
+      - name: Build package
+        if: ${{ !startsWith(inputs.gitref, 'v') }}
+        run: |
+          echo "::error Non-release deployment currently not supported"
+          exit 1
+
+      # Configure AWS communication via SSM.
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with: 
+          aws-region: eu-west-1
+          role-to-assume: "arn:aws:iam::${{ secrets[env.account-id] }}:role/${{ env.oidcrole }}-GithubActionsRole"
+          role-session-name: GithubActionsSession
+          
+      - name: Install awscli
+        uses: ./.github/actions/ssm_execute
+        with:
+          instance_id: ${{ secrets[env.instance-id] }}
+          command: |
+            sudo apt udpate; \
+            sudo apt install awscli -y
+
+      # Move packages to instance using S3.
+      # TODO: does this fail if the files already exist?
+      - name: Upload packages to S3
+        run: |
+          aws s3 cp ${{ env.node-package }} ${{ env.s3-path }}/${{ env.node-package }}
+          aws s3 cp ${{ env.node-faucet }} ${{ env.s3-path }}/${{ env.node-faucet }}
+
+      # TODO: does this fail if the files already exist?
```

Comment:
Do we need the still as a ToDo?

Reply:
Yeah its something I'll have to try out at some stage - can't try it locally since its a command on a remote aws server.